### ISSUE TITLE
Deploying with external postgres fails

### ISSUE THREAD

**Issue by jchanam** [goharbor/harbor-helm#99]
I'm trying to deploy the chart with an external postgres database and it makes all services to fail. (I'm using the branch 0.3.0)

I've checked and I've created the databases following the files that are used on the official db docker image found here https://github.com/goharbor/harbor/tree/master/make/photon/db.

I'm getting the following logs:
**adminserver**: `2018-11-28T16:44:38Z [FATAL] [main.go:46]: failed to initialize the system: pq: pg_hba.conf rejects connection for host "100.96.4.47", user "postgres", database "registry", SSL off`

**clair**: `{"Event":"pgsql: could not open database: pq: pg_hba.conf rejects connection for host \"100.96.4.46\", user \"postgres\", database \"clair\", SSL off","Level":"fatal","Location":"main.go:96","Time":"2018-11-28 16:49:08.544334"}`

**notary-server**: `waiting for postgres://postgres:<correct-password>@<postgres-endpoint>:5432/notaryserver?sslmode=disable to come up.`

**notary-signer**: `waiting for postgres://postgres:<correct-password>@<postgres-endpoint>:5432/notaryserver?sslmode=disable to come up.`

The connection strings are ok. I've tried manually and I can connect. Is there any preparation that I have to do on the postgres databases?

**Comment by ywk253100** [comment]
Could you try the image `goharbor/harbor-db:v1.6.0` to set up the external database?

**Comment by jchanam** [comment]
I can't use that image because I'm using the [postgres-operator](https://github.com/zalando-incubator/postgres-operator) to have high-availability databases.

I'd be awesome if an initialization sql files would be available to use, reachable from the documentation.

**Comment by ywk253100** [comment]
The chart on branch 0.3.0 uses the Harbor 1.6.0, so if you want to build the image by yourself you should use the code of branch 1.6.0 https://github.com/goharbor/harbor/tree/release-1.6.0/make/photon/db. The init sql is here https://github.com/goharbor/harbor/blob/release-1.6.0/make/migrations/postgresql/0001_initial_schema.up.sql.

**Comment by jchanam** [comment]
@ywk253100 I've used this three sql files to create the databases, as you're doing in the harbor-db image:
* https://github.com/goharbor/harbor/blob/master/make/photon/db/initial-notaryserver.sql
* https://github.com/goharbor/harbor/blob/master/make/photon/db/initial-notarysigner.sql
* https://github.com/goharbor/harbor/blob/master/make/photon/db/initial-registry.sql

Also, I've imported the initial schema into the registry database.

The logs are showing they can't connect to the database, but I do an exec to the pods, and I can connect with `psql`. What I'm doing wrong? Is there ANY other configuration to be done?

My databases:
```
                                   List of databases
     Name     |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges   
--------------+----------+----------+-------------+-------------+-----------------------
 clair        | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =Tc/postgres         +
              |          |          |             |             | postgres=CTc/postgres
 notaryserver | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =Tc/postgres         +
              |          |          |             |             | postgres=CTc/postgres+
              |          |          |             |             | server=CTc/postgres
 notarysigner | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =Tc/postgres         +
              |          |          |             |             | postgres=CTc/postgres+
              |          |          |             |             | signer=CTc/postgres
 postgres     | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | 
 registry     | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =Tc/postgres         +
              |          |          |             |             | postgres=CTc/postgres
```

**Comment by jchanam** [comment]
Hi @ywk253100,

I've found out that the issue is with the ssl mode for postgres. If set to "require", all pieces can connect except the adminserver. The issue is fixed with https://github.com/goharbor/harbor/pull/5861, but no image is released after v1.6.2 including this fix.

When a new release of harbor images will be done? When a new release with the chart will be done?

**Comment by ywk253100** [comment]
A new release will be available in late December if everything goes well.
